hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -156,9 +157,17 @@ def __tree_is_exhausted(self):
 
     def __stoppable_test_function(self, data):
         """Run ``self._test_function``, but convert a ``StopTest`` exception
-        into a normal return.
+        into a normal return and avoid raising Flaky for RecursionErrors.
         """
+        depth = stack_depth_of_caller()
+        # Because we add to the recursion limit, to be good citizens we also add
+        # a check for unbounded recursion.  The default limit is 1000, so this can
+        # only ever trigger if something really strange is happening and it's hard
+        # to imagine an intentionally-deeply-recursive use of this code.
+        assert depth <= 1000, "we probably shouldn't add to the stack limit here"
+        limit = sys.getrecursionlimit()
```

Comment:
Would it make sense to get the recursion limit just once, at the point of entry to the test, and use that? This helps us keep to a stable value throughout the test's execution.